cluster.go

```diff
@@ -147,8 +154,14 @@ type ClusterConfig struct {
 
 	// Dialer will be used to establish all connections created for this Cluster.
 	// If not provided, a default dialer configured with ConnectTimeout will be used.
+	// Dialer is ignored if HostDialer is provided.
 	Dialer Dialer
 
+	// HostDialer will be used to establish all connections for this Cluster.
+	// To support shard-aware port, HostDialer should implement ShardDialer.
+	// If not provided, Dialer will be used instead.
+	HostDialer HostDialer
```

Comment:
It's probably worth noting in the comment that a custom `HostDialer` must handle wrapping connection in TLS itself . There was no such requirement for `Dialer`, and the similar name might suggest that it's not required for `HostDialer` either.

Reply:
Good point! It is documented in the HostDialer interface, but mentioning it here will be useful too. Added.